packages/eslint-plugin/src/rules/space-infix-ops.ts

```diff
@@ -0,0 +1,153 @@
+     * @param {string} op The token to find
+     * @returns {TSESTree.Token|null}
+     * @private
+     */
+    function getFirstNonSpacedToken(
+      left: TSESTree.Token,
+      right: TSESTree.Token,
+      op: string,
+    ): TSESTree.Token | null {
+      const operator = sourceCode.getFirstTokenBetween(
+        left,
+        right,
+        token => token.value === op,
+      );
+      const prev = sourceCode.getTokenBefore(operator!);
+      const next = sourceCode.getTokenAfter(operator!);
+
+      if (
+        !sourceCode.isSpaceBetweenTokens(prev!, operator!) ||
+        !sourceCode.isSpaceBetweenTokens(operator!, next!)
+      ) {
+        return operator;
+      }
+
+      return null;
+    }
+
+    /**
+     * Reports an AST node as a rule violation
+     * @param {TSESTree.Node} mainNode The node to report
+     * @param {TSESTree.Token} culpritToken The token which has a problem
+     * @returns {void}
+     * @private
+     */
+    function report(
+      mainNode: TSESTree.Node,
+      culpritToken: TSESTree.Token,
+    ): void {
+      context.report({
+        node: mainNode,
+        loc: culpritToken.loc,
+        messageId: 'missingSpace',
+        data: {
+          operator: culpritToken.value,
+        },
+        fix(fixer) {
+          const previousToken = sourceCode.getTokenBefore(culpritToken);
+          const afterToken = sourceCode.getTokenAfter(culpritToken);
+          let fixString = '';
+
+          if (culpritToken.range[0] - previousToken!.range[1] === 0) {
+            fixString = ' ';
+          }
+
+          fixString += culpritToken.value;
+
+          if (afterToken!.range[0] - culpritToken.range[1] === 0) {
+            fixString += ' ';
+          }
+
+          return fixer.replaceText(culpritToken, fixString);
+        },
+      });
+    }
+
+    /**
+     * Check if it has an assignment char and report if it's faulty
+     * @param {TSESTree.Node} node The node to report
+     * @returns {void}
+     * @private
+     */
+    function checkForAssignmentSpace(node: TSEnumMember): void {
+      if (!node.initializer) {
+        return;
+      }
+
+      const leftNode = sourceCode.getTokenByRangeStart(
+        sourceCode.getIndexFromLoc(node.id.loc.start),
+      )!;
+      const rightNode = sourceCode.getTokenByRangeStart(
+        sourceCode.getIndexFromLoc(node.initializer.loc.start),
+      )!;
+
+      if (!rightNode) {
+        return;
+      }
+
+      const operator = '=';
+
+      const nonSpacedNode = getFirstNonSpacedToken(
+        leftNode,
+        rightNode,
+        operator,
+      );
+
+      if (nonSpacedNode) {
+        report(node as TSESTree.Node, nonSpacedNode);
+      }
+    }
+
+    const nodesToCheck = [AST_NODE_TYPES.TSEnumMember].reduce<
+      TSESLint.RuleListener
+    >((acc, node) => {
+      acc[node as string] = checkForAssignmentSpace;
+      return acc;
+    }, {});
+
+    return {
+      ...rules,
+      ...nodesToCheck,
```

Comment:
is there a reason you are doing the reduce thing with a single element? might be clearer if you do it like this instead:
```suggestion
    return {
      ...rules,
      [AST_NODE_TYPES.TSEnumMember]: checkForAssignmentSpace,
```
or simpler just write the function in place like:

```ts
return {
    ...rules,
    TSEnumMember(node): void {
      // content of above function here
    },
```

Reply:
Yeah i tried to take inspiration on other rules and kept it after trial and error. I'll fix it